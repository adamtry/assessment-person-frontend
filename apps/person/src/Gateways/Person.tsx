import axios from "axios";
import {
  personSearchResult,
} from "../Interfaces";
import { getToken } from "../Utils/getHackneyToken";

export const getPerson = async (id: string): Promise<personSearchResult | undefined> => {
  try {
    let response = await axios.get(
      `${process.env.API_URL}/residents/all`,
      {
        headers: {
          Authorization: `${getToken()}`,
        },
      }
    );

    var allPeople = response.data as personSearchResult[];
    var person = allPeople.find((person) => person.id == id);

    if (response?.status != 200) {
      throw new Error("Error retrieving person");
    }

    return person;
  } catch (e) {
    return undefined;
  }
};
