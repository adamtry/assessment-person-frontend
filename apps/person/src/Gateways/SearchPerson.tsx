import axios from "axios";
import { getToken } from "../Utils/getHackneyToken";
import { personSearchResult } from "../Interfaces";

interface params {
  query: string;
}

export const SearchPerson = async (
  params: params
): Promise<personSearchResult[]> => {
  let requestUrl = `${process.env.API_URL}/residents/search?query=${params.query}`;
  const response = await axios.get(requestUrl, {
    headers: {
      authorization: `${getToken()}`,
    },
  });

  const results: personSearchResult[] = response.data;

  return results;
};
