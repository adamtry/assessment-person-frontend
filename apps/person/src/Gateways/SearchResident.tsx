import axios from "axios";
import { getToken } from "../Utils/getHackneyToken";
import { personSearchResult } from "../Interfaces";

interface params {
  firstName: string;
  lastName: string;
  address: string | null;
  jigsawToken: string | null;
  dateOfBirth: string | null;
}

export const SearchPerson = async (
  params: params
): Promise<personSearchResult[]> => {
  let requestUrl = `${process.env.API_URL}/residents/search?query=${params.firstName}`;
  const response = await axios.get(requestUrl, {
    headers: {
      authorization: `${getToken()}`,
    },
  });

  const results: personSearchResult[] = response.data;

  return results;
};
