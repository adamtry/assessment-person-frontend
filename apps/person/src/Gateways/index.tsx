export * from "./GetPerson";
export * from "./SearchPerson";
