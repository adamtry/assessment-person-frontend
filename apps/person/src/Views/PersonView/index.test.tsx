import React from 'react'
import { render } from "@testing-library/react";
import {PersonView} from ".";

import * as GetPerson from "../../Gateways/GetPerson";
import { MemoryRouter } from 'react-router-dom';

beforeEach(() => {
    jest.resetModules();

    jest.mock("axios", () => {
        return {
            get: jest.fn(),
        };
    });

    jest.spyOn(GetPerson, 'getPerson').mockImplementation(() => {
        return Promise.resolve({
            id: "1",
            firstName: "Test",
            lastName: "User",
            email: "test.user@email.com",
            gender: "None",
        });
    });
});

describe("Person View", () => {

    test("should render person view", () => {
        const wrapper = render(
            <MemoryRouter>
                <PersonView />
            </MemoryRouter>
        );
        expect(wrapper).toMatchSnapshot();
    });
});