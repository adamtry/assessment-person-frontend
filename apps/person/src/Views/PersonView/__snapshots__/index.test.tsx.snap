// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person View should render person view 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="govuk-tabs lbh-tabs sv-space-t"
        data-module="govuk-tabs"
      >
        <h2
          class="govuk-tabs__title"
        >
          Contents
        </h2>
        <div
          class="govuk-!-margin-bottom-5"
        >
          <a
            class="govuk-back-link lbh-link lbh-link--no-visited-state"
            href="/search"
            id="back-to-search"
          >
            Back to search results
          </a>
          <a
            class="govuk-link lbh-link lbh-link--no-visited-state govuk-!-margin-left-2"
            href="/"
            id="new-search"
          >
            New search
          </a>
        </div>
        <h1>
           
        </h1>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key govuk-!-width-one-half"
          >
            <h5
              class="lbh-heading-h5"
            >
              Email
            </h5>
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            <div
              class="lbh-body-s sv-null-field"
              data-testid="emailField"
            >
              Not known
            </div>
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key govuk-!-width-one-half"
          >
            <h5
              class="lbh-heading-h5"
            >
              Gender
            </h5>
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            <div
              class="lbh-body-s sv-null-field"
              data-testid="genderField"
            >
              Not known
            </div>
          </dd>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="govuk-tabs lbh-tabs sv-space-t"
      data-module="govuk-tabs"
    >
      <h2
        class="govuk-tabs__title"
      >
        Contents
      </h2>
      <div
        class="govuk-!-margin-bottom-5"
      >
        <a
          class="govuk-back-link lbh-link lbh-link--no-visited-state"
          href="/search"
          id="back-to-search"
        >
          Back to search results
        </a>
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state govuk-!-margin-left-2"
          href="/"
          id="new-search"
        >
          New search
        </a>
      </div>
      <h1>
         
      </h1>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key govuk-!-width-one-half"
        >
          <h5
            class="lbh-heading-h5"
          >
            Email
          </h5>
        </dt>
        <dd
          class="govuk-summary-list__value"
        >
          <div
            class="lbh-body-s sv-null-field"
            data-testid="emailField"
          >
            Not known
          </div>
        </dd>
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key govuk-!-width-one-half"
        >
          <h5
            class="lbh-heading-h5"
          >
            Gender
          </h5>
        </dt>
        <dd
          class="govuk-summary-list__value"
        >
          <div
            class="lbh-body-s sv-null-field"
            data-testid="genderField"
          >
            Not known
          </div>
        </dd>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
